lib/interpreters/periodic_visits.rb

```diff
@@ -74,7 +74,7 @@ def generate_timewindows(timewindows_set)
         if @have_day_index
           first_day = timewindow.day_index ? (@schedule_start..@schedule_end).find{ |day| day % 7 == timewindow.day_index } : @schedule_start
           (first_day..@schedule_end).step(timewindow.day_index ? 7 : 1).collect{ |day_index|
-            Models::Timewindow.new(start: (timewindow.start || 0) + day_index * 86400,
+            Models::Timewindow.new(start: timewindow.start + day_index * 86400,
                                    end: (timewindow.end || 86400) + day_index * 86400)
```

Comment:
You are right. I do not see what could break if timewindow has default end.
And in fact I think even in general context we could set default timewindow to 86400. But you are right, in some specific cases this could be a problem.

If you are okay with that I will create a ticket for that so that we can merge this PR and think better about consequences of assigning default timewindow end.

Reply:
Sure ticket is alright, thanks Adeline ! You can give the the link to [this conversation](https://github.com/Mapotempo/optimizer-api/pull/143#discussion_r587367128) to have the context.

> And in fact I think even in general context we could set default timewindow to 86400. But you are right, in some specific cases this could be a problem.

Multi-day VRP is still general for me but in any case the "problem" is big enough (i.e., the user has to define TW.ends, even if they don't have any time-windows) that we cannot do it. Plus in the other simpler general cases, we might have some logic inside the wrappers that depend on `tw.end.nil?` at the moment (we have to check) but for the periodic case the initialiser will be :ok_hand: hopefully.